packages/unravel/app/blog/[id]/page.tsx

```diff
@@ -1,7 +1,17 @@
+import { z } from "zod";
+import { getBlog } from "../blog-data";
+
 interface Props {
   params: { id: string };
 }
 
-export default function BlogPost({ params: { id } }: Props) {
-  return <h1>Blog Post {id}</h1>;
+export default async function BlogPost({ params: { id } }: Props) {
```

Comment:
Don't need to solve now, but I wonder how we can include the title in the route? `/blog/xhfhkspaj` isn't a nice URL, but we ofc need the rkey for lookup.

Couple of ideas:

 `/blog/[rkey]/[slug]` eg. `/blog/xxxxx/hello-world`. We'd need to lookup rkey, and then check `slug == slugify(record.title)`, otherwise we `notFound()`.

Could also do the same thing but with `/blog/${rkey}-${slug}` eg. `/xxxx-hello-world`.

Reply:
I quite like the first idea

Reply:
Aren't we fetching these ourselves and storing them in the PDS on creation anyway? Can always  just do /blog/[slug] at that point and grab from our PDS.

Reply:
We couldn't use `getRecord` then. We'd have to fetch all content with `listRecords), potentially paginate, do an in-memory search for the one we want.

Reply:
Bust out the flow charts